homeassistant/components/playstation_network/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""The Playstation Network integration."""
+
+from __future__ import annotations
+
+from psnawp_api.core.psnawp_exceptions import PSNAWPAuthenticationError
+from psnawp_api.psn import PlaystationNetwork
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_NPSSO
+from .coordinator import PlaystationNetworkCoordinator
+
+PLATFORMS: list[Platform] = [Platform.MEDIA_PLAYER]
+
+
+type PlaystationNetworkConfigEntry = ConfigEntry[PlaystationNetworkCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: PlaystationNetworkConfigEntry
+) -> bool:
+    """Set up Playstation Network from a config entry."""
+
+    try:
+        psn = PlaystationNetwork(entry.data[CONF_NPSSO])
```

Comment:
```suggestion
        psn = PlaystationNetwork(entry.data[CONF_NPSSO])
```
this should go before the try block